packages/next/src/server/web/spec-extension/unstable-cache.ts

```diff
@@ -152,54 +152,51 @@ export function unstable_cache<T extends Callback>(
       if (workStore) {
         workStore.nextFetchId = fetchIdx + 1
 
-        const isNestedCache =
-          workUnitStore &&
-          (workUnitStore.type === 'unstable-cache' ||
-            workUnitStore.type === 'cache')
-
         // We are in an App Router context. We try to return the cached entry if it exists and is valid
         // If the entry is fresh we return it. If the entry is stale we return it but revalidate the entry in
         // the background. If the entry is missing or invalid we generate a new entry and return it.
 
         // We update the store's revalidate property if the option.revalidate is a higher precedence
-        if (!isNestedCache) {
+        if (
+          workUnitStore &&
+          (workUnitStore.type === 'cache' ||
+            workUnitStore.type === 'prerender' ||
+            workUnitStore.type === 'prerender-legacy')
+        ) {
+          // options.revalidate === undefined doesn't affect timing.
+          // options.revalidate === false doesn't shrink timing. it stays at the maximum.
           if (typeof options.revalidate === 'number') {
-            if (
-              typeof workStore.revalidate === 'number' &&
-              workStore.revalidate < options.revalidate
-            ) {
+            if (workUnitStore.revalidate < options.revalidate) {
               // The store is already revalidating on a shorter time interval, leave it alone
             } else {
-              workStore.revalidate = options.revalidate
+              workUnitStore.revalidate = options.revalidate
             }
-          } else if (
-            options.revalidate === false &&
-            typeof workStore.revalidate === 'undefined'
```

Comment:
This is behavior change / fix. Because it used to be that we don't override if we already have a number or boolean so it's not consistently the same result based on a race condition between two calls. But if we assume we can only lower, then a `false` value never lowers.